src/test/java/seleautomation/com/HalaTraining1/GitHubtest.java

```diff
@@ -0,0 +1,38 @@
+package seleautomation.com.HalaTraining1;
+
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import utilities.DataReader;
+import utilities.DriverFactory;
+
+public class GitHubtest {
+
+    @DataProvider(name = "testData")
+    public Object[][] provideTestData() {
+        DataReader reader = new DataReader("\"C:\\Users\\MASS\\OneDrive\\Desktop\\QA.xlsx\""); 
+        return reader.readTestData();
+    }
+
+    @Test(dataProvider = "testData")
+    public void testSignIn(String testCaseId, String username, String password, boolean expectedResult) {
+        WebDriver driver = DriverFactory.createDriver("chrome"); 
+
+        GitHubHomePage homePage = new GitHubHomePage(driver);
+        GitHubSignInPage signInPage = homePage.navigateToSignIn();
+
+        signInPage.signIn(username, password);
+
+        // Assert sign-in results
+        if (expectedResult) {
+            Assert.assertTrue(signInPage.isSignInSuccessful(), "Sign-in failed unexpectedly.");
```

Comment:
What is the differences between the if statements , for which cases ? 
all you need just 2 if statements instead of the 3